lib/Conversion/StaticLogicToCalyx/StaticLogicToCalyx.cpp

```diff
@@ -48,70 +48,30 @@ namespace staticlogictocalyx {
 /// Calyx component.
 using FuncMapping = DenseMap<FuncOp, calyx::ComponentOp>;
 
-/// A structure that abstracts a WhileOp or PipelineWhileOp to provide a
-/// consistent interface.
-struct WhileOpInterface {
-  WhileOpInterface(Operation *op) {
-    assert(isSupported(op));
-    if (auto scfWhile = dyn_cast<scf::WhileOp>(op))
-      impl = scfWhile;
-    if (auto staticLogicWhile = dyn_cast<staticlogic::PipelineWhileOp>(op))
-      impl = staticLogicWhile;
-  }
-
-  static bool isSupported(Operation *op) {
-    return isa<scf::WhileOp, staticlogic::PipelineWhileOp>(op);
-  }
+class StaticLogicWhileOp
+    : public calyx::WhileOpInterface<staticlogic::PipelineWhileOp> {
+public:
+  explicit StaticLogicWhileOp(Operation *op)
```

Comment:
I think this can be restricted to `explicit StaticLogicWhileOp(staticlogic::PipelineWhileOp op)` now.